test/exe/show_version_test.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -e -o pipefail
+
+source/exe/envoy-static --version |grep -E '[0-9]+\.[0-9]+\.[0-9]+'
```

Comment:
You can probably get the version from a `genrule` pointing at https://github.com/envoyproxy/data-plane-api/blob/master/docs/root/intro/version_history.rst (with `grep/head/sed`), this input would come from the `@envoy_api//docs/root/intro:version_history.rst` input.

Or, you could at least change this test to validate consistency. As it is today, we have two places in the `envoy` and `data-plane-api` changes that version is maintained after this PR, would be ideal to have single-source-of-truth or keep them consistent.

Reply:
Oh, I also found `docs/conf.py` in data-plane-api also has the version string (`1.6.0` there).

Considering the above and https://github.com/envoyproxy/envoy/pull/2276#discussion_r158964961, we should have single version string at `VERSION` file in data-plane-api repo, and make use of it at `docs/conf.py` and `VersionInfo` object. For `version_history.rst` and `RAW_RELEASE_NOTES.md`, since it's difficult to embed the version string from `VERSION` file, we can just write tests which validate consistency.

If it looks good, I can work for that. My concern is it's too complex or not for the task. Should we just hard code the version in a few places?

Also I suggest the version string will be next release version + `.dev` (e.g. `1.6.0.dev` now) to indicate "the git SHA doesn't match the original SHA for the release version".

Want any comments.

Reply:
I think this is the best solution, go for it.

Reply:
Yes +1. Let's move the version out into a dedicated file, and read into docs conf, etc. Thank you!!!